### PR TITLE
Disable AppArmor in generic-aarch64

### PR DESCRIPTION
**Pull request by lmbarros**
This aligns the generic-aarch64 kernel to the generic-amd64 kernel with regards to AppArmor configuration.

Initial motivation for this change comes balenaEngine: newer releases of Moby expect the AppArmor userland tools to be installed if AppArmor is enabled on the kernel. Given that balenaOS doesn't distribute the AppArmor tools, our aarch64 tests running on CI were failing for new Engine releases that incorporate those Moby changes.

### CODE REVIEW

File: layers/meta-balena-generic/recipes-kernel/linux/files/generic-aarch64/defconfig [hunk 1]
```diff
@@ -2995,15 +2995,18 @@ CONFIG_FORTIFY_SOURCE=y
...
 CONFIG_SECURITY_LOCKDOWN_LSM=y
 CONFIG_SECURITY_LOCKDOWN_LSM_EARLY=y
 CONFIG_INTEGRITY_SIGNATURE=y
 CONFIG_INTEGRITY_ASYMMETRIC_KEYS=y
 # CONFIG_INTEGRITY_TRUSTED_KEYRING is not set
 CONFIG_INTEGRITY_PLATFORM_KEYRING=y
-CONFIG_DEFAULT_SECURITY_APPARMOR=y
-CONFIG_LSM="lockdown,yama,loadpin,safesetid,integrity,apparmor,selinux,smack,tomoyo"
+# CONFIG_DEFAULT_SECURITY_APPARMOR is not set
+CONFIG_LSM="yama,loadpin,safesetid,integrity"
```
Comment (alexgg) [inline layers/meta-balena-generic/recipes-kernel/linux/files/generic-aarch64/defconfig:3009]: `lockdown` needs to be kept - we use this feature in secure boot

**Review by alexgg — CHANGES_REQUESTED** [review]

**Comment by alexgg** [comment]
> @alexgg, @jakogut I pushed a commit adding `lockdown` back to to aarch64. Also took the chance to add it to amd64, given it wasn't there previously. (I am assuming that's the right thing to do, let me know otherwise).
> 
> I also tested this PR locally, and it solved the AppArmor issue we were having with the next Engine release. So, I am marking this as ready for review.

hi @lmbarros, strange that this was not defined in amd64. We define `SECURITY_LOCKDOWN_LSM` and `SECURITY_LOCKDOWN_LSM_EARLY` in https://github.com/balena-os/meta-balena/blob/707ed18ca071fb7d7d557e0774f9f479323892af/meta-balena-common/classes/kernel-balena.bbclass#L667, and the `CONFIG_LSM` help states:

```
255           A comma-separated list of LSMs, in initialization order.              
256           Any LSMs left off this list will be ignored.
```

@mtoman @jakogut I think we should add tests for kernel lockdown in secure boot including:

* Modifying kernel boot parameters will not boot 
* Unsigned kernel modules will not be loaded

**Comment by alexgg** [comment]
@lmbarros could you print the differences between the kernel configurations before and after the changes? Looking at the defconfig only is not that helpful as we apply changes in the kernel class too. Comparing the final .config files will show the exact differences.

**Comment by alexgg** [comment]
> You mean, zcat /proc/config.gz before and after the changes or is there some quicker way?

That's one way. You can also just rebuilt the `virtual/kernel` package with and without the changes and compare the config files in `tmp/work-shared/generic-amd64/kernel-build-artifacts/.config`

>  but this whole PR was meant to be just a simple fix to unblock my work on the Engine

I understand, but kernel configuration changes need to be done with care

**Comment by lmbarros** [comment]
@alexgg , here's the diff you asked for between the kernel configurations before and after the changes:

```
 # CONFIG_IMA_KEYRINGS_PERMIT_SIGNED_BY_BUILTIN_OR_SECONDARY is not set
 # CONFIG_IMA_SECURE_AND_OR_TRUSTED_BOOT is not set
 # CONFIG_EVM is not set
-# CONFIG_DEFAULT_SECURITY_SELINUX is not set
+CONFIG_DEFAULT_SECURITY_SELINUX=y
 # CONFIG_DEFAULT_SECURITY_TOMOYO is not set
-CONFIG_DEFAULT_SECURITY_APPARMOR=y
+# CONFIG_DEFAULT_SECURITY_APPARMOR is not set
 # CONFIG_DEFAULT_SECURITY_DAC is not set
-CONFIG_LSM="lockdown,yama,loadpin,safesetid,integrity,apparmor,selinux,smack,tomoyo"
+CONFIG_LSM="lockdown,yama,loadpin,safesetid,integrity"
```

**Comment by lmbarros** [comment]
(That's for aarch64, BTW)

**Comment by alexgg** [comment]
@lmbarros left a comment in https://balena.zulipchat.com/#narrow/stream/345889-loop.2Fbalena-os/topic/Kernel.20LSM.20settings

**Comment by lmbarros** [comment]
Superseded by https://github.com/balena-os/balena-generic/pull/214 and https://github.com/balena-os/meta-balena/pull/2996/files.

**Pull request closed without merging**